packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -1239,6 +1239,50 @@ class PostsRepo extends AbstractRepo<"Posts"> {
       };
     });
   }
+
+  /**
+   * Get posts from users the current user is subscribed to, for UltraFeed
+   */
+  async getPostsFromSubscribedUsersForUltraFeed(
+    context: ResolverContext,
+    maxAgeDays: number,
+    limit = 100
+  ): Promise<{ postId: string }[]> {
+    const db = this.getRawDb();
+    const userId = context.currentUser?._id;
+
+    if (!userId) {
+      return [];
+    }
+
+    return await db.manyOrNone<{ postId: string }>(`
+      -- PostsRepo.getPostsFromSubscribedUsersForUltraFeed
+      WITH user_subscriptions AS (
```

Comment:
I don't think this needs to be in its own CTE and expect that to just make the query slower, since it requires strictly more (and more sequential) work from the database than just doing a direct join.  (Worth double-checking, but I would be surprised if not.)

Reply:
My understanding is CTE means that all the results for the CTE expression must be returned, which can be expensive, but for subscriptions seems unlikely to be that many or much of a problem